internal/db/schema/migrations/postgres/14/01_event.up.sql

```diff
@@ -0,0 +1,341 @@
+create function delete_event_sink_subtype()
+    returns trigger
+as $$
+begin
+    delete from event_sink
+    where public_id = old.public_id;
+    return null; -- result is ignored since this is an after trigger
+end;
+$$ language plpgsql;
+
+create table event_file_sink(
+    public_id wt_public_id primary key,
+    config_id wt_public_id not null,
+    event_type text not null
+        constraint event_type_enm_fkey
+            references event_type_enm(name)
+            on delete restrict
+            on update cascade,
+    format_type text not null
+        constraint event_format_type_enm_fkey
+            references event_format_type_enm(name)
+            on delete restrict
+            on update cascade,
+    allow_filter wt_bexprfilter,
+    deny_filter wt_bexprfilter,
+    path text not null 
+        constraint path_not_empty
+            check (
+                length(trim(path)) > 0
+            ),
+    filename text not null
+        constraint filename_not_empty
+            check (
+                length(trim(filename)) > 0
+            ),
+    constraint path_filename_uq
+        unique(path, filename), -- ensure each sink is writing to a unique file
+    rotate_bytes int
+        constraint rotate_bytes_null_or_greater_than_zero
+            check(
+                rotate_bytes > 0
+            ),
+    rotate_duration interval
+        constraint rotate_duration_null_or_greater_than_zero
+            check(
+                rotate_duration > '0'::interval
+            ),
+    rotate_max_files int
+        constraint rotate_max_files_null_or_greater_than_zero
+            check(
+                rotate_max_files > 0
+            ),
+    constraint event_sink_fkey
+      foreign key(config_id, public_id)
+      references event_sink(config_id, public_id)
+      on delete cascade
+      on update cascade
+);
+
+comment on table event_file_sink is 
+'event_file_sink is a table where each entry represents a configured event file '
+'sink';
+
+create trigger
+  immutable_columns
+before
+update on event_file_sink
+  for each row execute procedure immutable_columns(
+      'public_id', 
+      'config_id'
+);
+
+create trigger
+    insert_event_sink_subtype 
+before insert on event_file_sink
+    for each row execute procedure insert_event_sink_subtype();
+
+create trigger
+    delete_event_sink_subtype 
+after delete on event_file_sink
+    for each row execute procedure delete_event_sink_subtype();
+
+create table event_stderr_sink(
+    public_id wt_public_id primary key,
+    config_id wt_public_id not null,
+    event_type text not null
+        constraint event_type_enm_fkey
+            references event_type_enm(name)
+            on delete restrict
+            on update cascade,
+    format_type text not null
+    constraint event_format_type_enm_fkey
+        references event_format_type_enm(name)
+        on delete restrict
+        on update cascade,
+    allow_filter wt_bexprfilter,
```

Comment:
I'm sorry, I didn't notice this earlier but both allow and deny filters are a list so they need to be in a separate table.